src/test/java/org/sagebionetworks/bridge/services/TemplateServiceTest.java

```diff
@@ -0,0 +1,575 @@
+        verify(mockCriteriaDao).getCriteria("template:guidOne");
+        verify(mockCriteriaDao).getCriteria("template:guidTwo");
+    }
+    
+    @Test
+    public void getTemplatesForTypeDefaultsCriteriaObject() {
+        Template t1 = Template.create();
+        t1.setGuid("guidOne");
+        
+        Template t2 = Template.create();
+        t2.setGuid("guidTwo");
+        
+        List<Template> list = ImmutableList.of(t1, t2);
+        PagedResourceList<? extends Template> resourceList = new PagedResourceList<>(list, 150);
+        doReturn(resourceList).when(mockTemplateDao).getTemplates(TEST_STUDY, EMAIL_RESET_PASSWORD, 5, 50, true);
+        
+        PagedResourceList<? extends Template> results = service.getTemplatesForType(TEST_STUDY, EMAIL_RESET_PASSWORD, 5, 50, true);
+        
+        for (Template template : results.getItems()) {
+            assertNotNull(template.getCriteria());
+            assertNotNull(template.getCriteria().getKey());
+        }
+    }
+    
+    @Test
+    public void getTemplatesForTypeDefaultsOffset() {
+        PagedResourceList<? extends Template> resourceList = new PagedResourceList<>(ImmutableList.of(), 150);
+        doReturn(resourceList).when(mockTemplateDao).getTemplates(TEST_STUDY, EMAIL_RESET_PASSWORD, 0, 50, true);
+        
+        service.getTemplatesForType(TEST_STUDY, EMAIL_RESET_PASSWORD, null, 50, true);
+        
+        verify(mockTemplateDao).getTemplates(TEST_STUDY, EMAIL_RESET_PASSWORD, 0, 50, true);
+    }
+
+    @Test
+    public void getTemplatesForTypeDefaultsPageSize() {
+        PagedResourceList<? extends Template> resourceList = new PagedResourceList<>(ImmutableList.of(), 150);
+        doReturn(resourceList).when(mockTemplateDao).getTemplates(TEST_STUDY, EMAIL_RESET_PASSWORD, 5, API_DEFAULT_PAGE_SIZE, true);
+        
+        service.getTemplatesForType(TEST_STUDY, EMAIL_RESET_PASSWORD, 5, null, true);
+        
+        verify(mockTemplateDao).getTemplates(TEST_STUDY, EMAIL_RESET_PASSWORD, 5, API_DEFAULT_PAGE_SIZE, true);
+    }
+
+    @Test(expectedExceptions = BadRequestException.class)
+    public void getTemplatesOffsetLessThanZero() {
+        service.getTemplatesForType(TEST_STUDY, EMAIL_RESET_PASSWORD, -5, null, true);
+    }
+
+    @Test(expectedExceptions = BadRequestException.class)
+    public void getTemplatesPageSizeBelowMin() {
+        service.getTemplatesForType(TEST_STUDY, EMAIL_RESET_PASSWORD, null, 0, true);
+    }
+
+    @Test(expectedExceptions = BadRequestException.class)
+    public void getTemplatesPageSizeAboveMax() {
+        service.getTemplatesForType(TEST_STUDY, EMAIL_RESET_PASSWORD, null, 1000, true);
+    }
+
+    @Test
+    public void getTemplate() {
+        Criteria criteria = Criteria.create();
+        criteria.setKey("template:"+GUID1); // this is persisted as part of criteria, not set on load
+        when(mockCriteriaDao.getCriteria("template:"+GUID1)).thenReturn(criteria);
```

Comment:
> this is persisted as part of criteria, not set on load

This may be logically true, but loadCriteria() has code to set the criteriaKey. Is this redundant code that should be removed? Or is it code that we need and that we should test it?

Reply:
So thinking about this further, we don't serialize the key in the JSON we return, and we always set it before persistence, so we don't really need the key set when loading. I removed this and a couple of lines in the test that are looking for it.